src/main/java/de/hysky/skyblocker/skyblock/museum/MuseumManager.java

```diff
@@ -0,0 +1,313 @@
+
+			if (resultButton.visible && resultButton.isHovered()) this.hoveredDonationButton = resultButton;
+		}
+
+		if (this.sortButton.active) {
+			int iconX = this.sortButton.getX() + (this.sortButton.getWidth() - 16) / 2;
+			int iconY = this.sortButton.getY() + (this.sortButton.getHeight() - 16) / 2;
+			ItemStack stack = ITEM_SORTER.getCurrentSortingItem();
+			context.drawItemWithoutEntity(stack, iconX, iconY);
+			this.sortButton.render(context, mouseX, mouseY, delta);
+		}
+
+		if (this.filterButton.active) {
+			int iconX = this.filterButton.getX() + (this.filterButton.getWidth() - 16) / 2;
+			int iconY = this.filterButton.getY() + (this.filterButton.getHeight() - 16) / 2;
+			ItemStack stack = ITEM_FILTER.getCurrentFilterItem();
+			context.drawItemWithoutEntity(stack, iconX, iconY);
+			this.filterButton.render(context, mouseX, mouseY, delta);
+		}
+
+		// Render the page flip buttons
+		if (this.prevPageButton.active) this.prevPageButton.render(context, mouseX, mouseY, delta);
+		if (this.nextPageButton.active) this.nextPageButton.render(context, mouseX, mouseY, delta);
+
+		this.searchField.render(context, mouseX, mouseY, delta);
+
+		this.drawTooltip(context, mouseX, mouseY);
+	}
+
+	public void drawTooltip(DrawContext context, int x, int y) {
+		// Draw the tooltip of the hovered result button if one is hovered over
+		if (this.hoveredDonationButton != null && !this.hoveredDonationButton.getDisplayStack().isEmpty()) {
+			ItemStack stack = this.hoveredDonationButton.getDisplayStack();
+			Identifier tooltipStyle = stack.get(DataComponentTypes.TOOLTIP_STYLE);
+
+			context.drawTooltip(TEXT_RENDERER, hoveredDonationButton.getItemTooltip(), x, y, tooltipStyle);
+		}
+	}
+
+	@Override
+	public boolean mouseClicked(double mouseX, double mouseY, int button) {
+		if (this.searchField.mouseClicked(mouseX, mouseY, button)) {
+			this.searchField.setFocused(true);
+			return true;
+		} else if (this.nextPageButton.mouseClicked(mouseX, mouseY, button)) {
+			CURRENT_PAGE++;
+			updateButtons();
+			return true;
+		} else if (this.prevPageButton.mouseClicked(mouseX, mouseY, button)) {
+			CURRENT_PAGE--;
+			updateButtons();
+			return true;
+		} else if (this.filterButton.mouseClicked(mouseX, mouseY, button)) {
+			return true;
+		} else if (this.sortButton.mouseClicked(mouseX, mouseY, button)) {
+			return true;
+		}
+
+		this.searchField.setFocused(false);
+		this.filterButton.setFocused(false);
+		this.sortButton.setFocused(false);
+		return false;
+	}
+
+	@Override
+	public boolean keyReleased(int keyCode, int scanCode, int modifiers) {
+		return super.keyReleased(keyCode, scanCode, modifiers);
+	}
+
+	@Override
+	public boolean charTyped(char chr, int modifiers) {
+		if (this.searchField.charTyped(chr, modifiers)) {
+			updateSearchResults(true);
+			return true;
+		}
+		return false;
+	}
+
+	@Override
+	public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+		if ((this.searchField.isActive() && keyCode == GLFW.GLFW_KEY_E)
```

Comment:
If the check for the `E` key is for opening/closing the inventory then I would read from the player's key bind as it could be different (also maybe ensure that this handles the esc key if necessary too?)